packages/fast-components-react-msft/app/app.tsx

```diff
@@ -200,45 +189,64 @@ export default class App extends React.Component<{}, AppState> {
         );
     }
 
-    private generateDesignSystem(): DesignSystem {
-        const designSystem: Partial<DesignSystem> = {
-            accentPalette: this.state.accentPalette,
-            neutralPalette: this.state.neutralPalette,
-            direction: this.state.direction,
-            backgroundColor: this.state.backgroundColor,
-            density: this.state.density,
-        };
-
-        return Object.assign({}, DesignSystemDefaults, designSystem);
-    }
-
     private handleUpdateDirection = (direction: Direction): void => {
         const newDir: Direction =
-            this.state.direction === Direction.ltr ? Direction.rtl : Direction.ltr;
+            direction === Direction.ltr ? Direction.rtl : Direction.ltr;
 
-        if (this.state.direction === newDir) {
+        if (this.state.designSystem.direction === newDir) {
             return;
         }
 
         this.setState({
-            direction: newDir,
+            designSystem: clone(
+                merge({}, this.state.designSystem, {
+                    direction: newDir,
+                })
+            ),
         });
     };
 
+    private getNeutralPallete(colorSource: string): string[] {
+        const color: ColorRGBA64 = parseColor(colorSource);
+        const hslColor: ColorHSL = rgbToHSL(color);
+        const augmentedHSLColor: ColorHSL = ColorHSL.fromObject({
+            h: hslColor.h,
+            s: hslColor.s,
+            l: 0.5,
+        });
+        return this.createColorPalette(hslToRGB(augmentedHSLColor));
+    }
+
     private handleUpdateTheme = (theme: ThemeName): void => {
         if (theme !== ThemeName.custom) {
             this.setState({
                 theme,
                 backgroundColor: theme === ThemeName.dark ? dark : light,
+                designSystem: clone(
+                    merge({}, this.state.designSystem, {
+                        backgroundColor: theme === ThemeName.dark ? dark : light,
```

Comment:
what are `dark` and `light`? I think this is where we would want to pull the background value from the current design-system

Reply:
nm - this should be a different PR